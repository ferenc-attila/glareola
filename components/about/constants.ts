export const ABOUT = {
    app: {
        title: 'About Glareola Application',
        description: [
            'Read rarity observations from birding.hu',
        ],
        authors: [
            'Attila Ferenc',
        ],
        version: '0.0.1',
        date: '2023-07-15',
        license: {
            shortName: 'MIT',
            link: 'https://opensource.org/license/mit/',
        },
    },
    mainImage: {
        title: 'Collared pratincole (Glareola pratincola)',
        author: 'Saswat Mishra',
        date: '2018-03-30',
        location: 'Nal Sarovar, Gujarat',
        license: {
            shortName: 'CC-BY-SA-4.0',
            link: 'https://creativecommons.org/licenses/by-sa/4.0/',
        }
    }
}
