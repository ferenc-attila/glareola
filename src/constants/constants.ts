export const SOURCES = {
    BIRDING_HU: 'birding.hu',
    BIRDING_SK: 'birding.sk',
    RAREBIRDS_HU: 'rarebirds.hu',
};

export const URLS = {
    BIRDING_HU_MAIN: 'http://birding.hu/index.php?page=&cid=az_elmult_14_nap_ritkasagai&per_page=100',
};

export const BUTTON_PROPS = {
    SIZE: 32,
    COLOR: 'white',
};

export const ICON_NAMES = {
    HOME: 'home',
    OBSERVATIONS: 'binoculars',
    INFO: 'info-circle',
    SETTINGS: 'tools',
    IMAGE: 'camera',
    BACK: 'backward',
    MAP: 'map-marked-alt',
};
