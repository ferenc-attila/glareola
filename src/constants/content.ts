export const ABOUT = {
    APP: {
        VERSION: '0.0.7',
        DATE: '2023-10-08',
        LICENSE: {
            SHORT_NAME: 'MIT',
            LINK: 'https://opensource.org/license/mit/',
        },
    },
    COVER_IMAGE: {
        SOURCE: 'https://upload.wikimedia.org/wikipedia/commons/4/4c/Collared_Pratincole_SM.jpg',
        LICENSE: {
            SHORT_NAME: 'CC-BY-SA-4.0',
            LINK: 'https://creativecommons.org/licenses/by-sa/4.0/',
        },
    },
    ICON_IMAGE: {
        SOURCE: 'https://upload.wikimedia.org/wikipedia/commons/9/91/Glareola_pratincola_in_Ambos%C3%A9lie_National_Park_kenya%2C_crop2.jpg',
        LICENSE: {
            SHORT_NAME: 'CC-BY-SA-3.0',
            LINK: 'https://creativecommons.org/licenses/by-sa/3.0/',
        },
    },
};
