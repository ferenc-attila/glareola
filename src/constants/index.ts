export { ABOUT } from './content';
export { BUTTON_PROPS, ICON_NAMES, SOURCES, URLS } from './constants';
