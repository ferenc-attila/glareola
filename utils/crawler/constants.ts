export const sources = {
    birdingHU: 'birding.hu',
    birdingSK: 'birding.sk',
    rareBirdsHU: 'rarebirds.hu',
}
