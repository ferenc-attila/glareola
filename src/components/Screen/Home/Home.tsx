import React from 'react';
import { Image, StyleSheet, Text } from 'react-native';

import { ABOUT } from '../../../constants';
import i18n from '../../../localization';
import { COLORS, TEXT_STYLES } from '../../../styles';

export const Home = () => (
    <>
        <Text style={styles.appName}>{i18n.t('APP_NAME')}</Text>
        <Image
            source={require('../../../assets/images/glareola_pratincola.png')}
            style={styles.titleImage}
            accessibilityRole='image'
            accessibilityLabel={i18n.t('COVER_IMAGE_DESCRIPTION')}
        />
        <Text style={styles.commonText}>{i18n.t('APP_DESCRIPTION')}</Text>
        <Text style={styles.smallText}>{ABOUT.APP.VERSION}</Text>
    </>
);

const styles = StyleSheet.create({
    appName: {
        paddingBottom: 25,
        color: COLORS.FONT.NORMAL,
        fontWeight: '900',
        fontSize: 40,
    },
    titleImage: {
        margin: 30,
        maxWidth: '90%',
        maxHeight: '50%',
        resizeMode: 'contain',
    },
    commonText: {
        ...TEXT_STYLES.COMMON,
    },
    smallText: {
        ...TEXT_STYLES.SMALL,
    },
});
