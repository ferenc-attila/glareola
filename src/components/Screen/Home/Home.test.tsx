import { render } from '@testing-library/react-native/build/pure';

import { Home } from './Home';
import { ABOUT } from '../../../constants';

describe('Home', () => {
    test('should render correctly', () => {
        const container = render(<Home />);
        expect(container.toJSON()).toMatchSnapshot();
    });

    test('image should have correct accessibility label', () => {
        const container = render(<Home />);
        expect(container.getByLabelText('Collared pratincole (Glareola pratincola)')).toBeTruthy();
    });

    test('version number should be correct', () => {
        const container = render(<Home />);
        expect(container.getByText(ABOUT.APP.VERSION)).toBeTruthy();
    });
});
