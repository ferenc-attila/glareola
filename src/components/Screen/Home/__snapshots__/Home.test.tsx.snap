// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
[
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 40,
        "fontWeight": "900",
        "paddingBottom": 25,
      }
    }
  >
    Glareola
  </Text>,
  <Image
    accessibilityLabel="Collared pratincole (Glareola pratincola)"
    accessibilityRole="image"
    source={1}
    style={
      {
        "margin": 30,
        "maxHeight": "50%",
        "maxWidth": "90%",
        "resizeMode": "contain",
      }
    }
  />,
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 14,
        "fontWeight": "500",
      }
    }
  >
    A crawler for rarity observations
  </Text>,
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 10,
        "fontWeight": "300",
      }
    }
  >
    0.0.8
  </Text>,
]
`;
